dyno-core/src/main/java/com/netflix/dyno/connectionpool/impl/lb/HostSelectionWithFallback.java

```diff
@@ -120,7 +123,7 @@ public HostSelectionWithFallback(ConnectionPoolConfiguration config, ConnectionP
 			try { 
 				return hostPool.borrowConnection(duration, unit);
 			} catch (DynoConnectException e) {
-				if (e instanceof PoolExhaustedException) {
+				if (e instanceof PoolExhaustedException && replicationFactor.get() == 3) {
```

Comment:
After reviewing again, I realized that bit of code actually introduces a logic issue. I'm going to remove the replication check and add a comment for clarification.